DataFormats/simulation/include/SimulationDataFormat/MCCompLabel.h

```diff
@@ -47,54 +49,84 @@ class MCCompLabel
   // mask for all used fields
   static constexpr ULong64_t maskFull = (ul0x1 << (nbitsTrackID + nbitsEvID + nbitsSrcID)) - 1;
 
-  MCCompLabel(int trackID, int evID = 0, int srcID = 0) { set(trackID, evID, srcID); }
-  MCCompLabel() = default;
+  MCCompLabel(int trackID, int evID, int srcID, bool fake) { set(trackID, evID, srcID, fake); }
+  MCCompLabel(bool invalid = false)
+  {
+    if (invalid) {
+      mLabel = Invalid;
+    } else {
+      mLabel = NotSet;
+    }
+  }
   ~MCCompLabel() = default;
 
   // check if label was assigned
   bool isSet() const { return mLabel != NotSet; }
   // check if label was not assigned
   bool isEmpty() const { return mLabel == NotSet; }
-  // check if label was assigned with non-negaive trackID
-  bool isPosTrackID() const { return getTrackID() >= 0; }
-  // return 1 if the tracks are the same and have labels>=0,
-  // 0 if the tracks are the same but at least one of them has label<0
+  // check if label corresponds to real particle
+  bool isValid() const { return mLabel != Invalid; }
+  // check if label was assigned as for correctly identified particle
+  bool isSetAndValid() const { return isSet() && isValid(); }
+
+  // check if label was assigned as for incorrectly identified particle
+  bool isFake() const { return mLabel & Fake; }
```

Comment:
With the current logic, `isFake()` will also be true if the label is unset and if the label is invalid, so the comment for the `isFake()` function is wrong.

Somehow, I think this is actually the correct behavior, since e.g, a track consisting of noise clusters (which will be invalid) should be considered fake. On the other hand, if the track is invalid or unset, getTrackId() will return a nonreasonable value.

I'd either add `&& isSetAndValid()` to `isFake()`, or change the comment for `isFake()` and change `isCorrect()` to only `return !isFake()`